include/sir/platform.h

```diff
@@ -402,6 +406,9 @@ _set_thread_local_invalid_parameter_handler(
 #   include <fcntl.h>
 #  endif
 #  include <unistd.h>
+#  if defined(__APPLE__) && defined(__MACH__)
```

Comment:
you can just replace this with `if defined(__MACOS__)` because those are the exact macros we check in order to define it